solr/contrib/blob-directory/src/java/org/apache/solr/blob/BlobStoreConnection.java

```diff
@@ -26,21 +29,30 @@
 
 import java.io.Closeable;
 import java.io.IOException;
-import java.io.InputStream;
 import java.io.OutputStream;
 import java.lang.invoke.MethodHandles;
 import java.net.URI;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
 import java.util.Objects;
-import java.util.concurrent.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 /**
- * Pushes a set of files to Blob, and works with listings.
+ * Connection to a {@link BackupRepository} to store persistently directories files.
+ * Provides support to:
+ * <ul>
+ *   <li>List and pull repository files.</li>
+ *   <li>Push local files to the repository.</li>
+ * </ul>
  */
-public class BlobPusher implements Closeable {
+public class BlobStoreConnection implements Closeable {
```

Comment:
Not a bad name but I'd prefer something better as well.  Since it works with the BackupRepository, the "Repository" word in there is important.  The "Connection" part I like the least; this isn't low level connection oriented.  Maybe simply *BlobRepository*?  After all, it's close to the repository end (vs the local dir).  It's Blob related.

Reply:
+1